packages/EasyStandard/docs/services/sniffs.md

```diff
@@ -0,0 +1,410 @@
+    {
+        $value1 = 2;
+        $value2 = 2;
+        $expectedClosure = static function () use ($value1, $value2): int {
+            $result = $value1 + $value2;
+
+            return $result + 0;
+        };
+
+        $actualResult = 2 + 2;
+
+        self::assertSame($expectedClosure(), $actualResult);
+    }
+
+    public function noTestMethod()
+    {
+        $expectedResult = 4;
+        $actualResult = 2 + 2;
+        self::assertSame($expectedResult, $actualResult);
+    }
+}
+```
+**Parameters**
+
+- `testMethodPrefix` - If the method name starts with this prefix, checks will be applied to it. Default value: `test`.
+- `testNamespace` - If the class namespace begins with this prefix, the class will be parsed. Default value: `App\Tests`.
+```php
+// correct
+namespace App\NoTestNamespace;
+
+final class TestClass
+{
+    public function testSomething()
+    {
+        $expectedResult = 4;
+
+        $actualResult = 2 + 2;
+
+        self::assertSame($expectedResult, $actualResult);
+
+        echo $actualResult;
+    }
+}
+```
+
+##### [\EonX\EasyStandard\Sniffs\ControlStructures\NoNotOperatorSniff][8]
+Strict comparison operator should be used instead NOT operator.
+```php
+// incorrect
+$a = (bool)\random_int(0, 1);
+if (!$a) {
+    // Do something.
+}
+````
+```php
+// correct
+$a = (bool)\random_int(0, 1);
+if ($a === false) {
+    // Do something.
+}
+````
+
+#### Exceptions
+
+##### [\EonX\EasyStandard\Sniffs\Exceptions\ThrowExceptionMessageSniff][9]
+Exception message must be either a variable or a translation message, started with one of valid prefixes.
+```php
+// incorrect
+throw new \Exception('Incorrect message');
+````
+```php
+// correct
+throw new NotFoundHttpException();
+// or
+$exception = new Exception('Some exception message');
+throw $exception;
+// or
+throw new InvalidArgumentException('exceptions.some_message');
+// or
+$message = 'Some exception message';
+throw new RuntimeException($message);
+````
+**Parameters**
+
+- `validPrefixes` - An array of prefixes to start with the message text line. Default value: `['exceptions.']`.
+
+#### Methods
+
+##### [\EonX\EasyStandard\Sniffs\Methods\TestMethodNameSniff][10]
+Checks that a method name matches/does not match a specific regex.
+**Parameters**
+
+- `allowed` - An array of regular expressions to match method names.
+Default value:
+```
+[
+    [
+        'namespace' => '/^App\\\Tests\\\Unit/',
+        'patterns' => ['/test[A-Z]/'],
+    ],
+]
+```
+- `forbidden` - An array of regular expressions that method names should not match.
+Default value:
+```
+[
+    [
+        'namespace' => '/^App\\\Tests\\\Unit/',
+        'patterns' => ['/(Succeed|Return|Throw)[^s]/'],
+    ],
+]
+```
+
+#### Namespaces
+
+##### [\EonX\EasyStandard\Sniffs\Namespaces\Psr4Sniff][11]
+Checks that namespace name match PSR-4 project structure.
+**Parameters**
+
+- `composerJsonPath` - Relative path to project file `composer.json`. Default value: `composer.json`.
```

Comment:
`A relative path to the project file...`

Reply:
Done.